python/ccf/proposal_generator.py

```diff
@@ -325,6 +325,15 @@ def set_jwt_public_signing_keys(issuer: str, jwks_path: str, **kwargs):
     return build_proposal("set_jwt_public_signing_keys", args, **kwargs)
 
 
+@cli_proposal
+def renew_node_certificate(node_id: str, valid_from: str, valid_to: str, **kwargs):
+    return build_proposal(
+        "renew_node_certificate",
+        {"node_id": node_id, "valid_from": valid_from, "valid_to": valid_to},
```

Comment:
I would like to revisit this and only have a `renew_node_certificates` proposal (with an optional `node_id` as argument). `valid_from` would be derived from the host untrusted time, and `valid_to` from the service-wide `node_cert_maximum_validity_period_days` configuration. I think this is much better from a usability perspective (similar to https://kubernetes.io/docs/tasks/administer-cluster/configure-upgrade-etcd/). 

The only drawback is that members wouldn't be able to review the untrusted time before it is set in the node certificates, which may cause some availability issues. Automatically adding the untrusted time as a proposal argument inside CCF sounds appealing (so that voters can review it), but this would break the governance history (signed request and proposal don't match).

Reply:
As discussed with @achamayou, it is preferable to stick to the manual `valid_to`/`valid_from` proposal arguments and avoid having the primary node implicitly use its untrusted time to determine the new validity of the node certificate. Otherwise, one malicious host could DOS the entire service (by setting the untrusted host time to far in the past/future). Additionally, the constitution may decide to skip member votes for that proposal, which means that no member could verify the validity of the host time, even if this one was included in the proposal.

A better future alternative would be to refresh the quote, and use the collateral time (which isn't extractable just yet, see #59) to deduce the certificate validity period. 

In any case, certificate refresh being such an infrequent event that the setting the validity period manually in the certificate is a fair trade-off.